scripts/test-v1-v2-rollover.ts

```diff
@@ -0,0 +1,149 @@
+        method: "hardhat_impersonateAccount",
+        params: [WHALE],
+    });
+
+    await hre.network.provider.request({
+        method: "hardhat_impersonateAccount",
+        params: [BORROWER],
+    });
+
+    await hre.network.provider.request({
+        method: "hardhat_impersonateAccount",
+        params: [LENDER],
+    });
+
+    const borrower = await hre.ethers.getSigner(BORROWER);
+    const lender = await hre.ethers.getSigner(LENDER);
+    const whale = await hre.ethers.getSigner(WHALE);
+
+    const erc20Factory = await ethers.getContractFactory("ERC20");
+    const usdc = <ERC20>await erc20Factory.attach(USDC_ADDRESS);
+
+    const erc721Factory = await ethers.getContractFactory("ERC721");
+    const note = <PromissoryNote>await erc721Factory.attach(BORROWER_NOTE_ADDRESS);
+
+    console.log("Deploying rollover...");
+
+    const factory = await ethers.getContractFactory("FlashRolloverV1toV2")
+    const flashRollover = <FlashRolloverV1toV2>await factory.deploy(ADDRESSES_PROVIDER_ADDRESS);
+
+    console.log("Doing approvals...");
+
+    await whale.sendTransaction({ to: borrower.address, value: ethers.utils.parseEther("100") });
+    await whale.sendTransaction({ to: lender.address, value: ethers.utils.parseEther("100") });
+    await whale.sendTransaction({ to: newLender.address, value: ethers.utils.parseEther("100") });
+    await usdc.connect(whale).transfer(newLender.address, ethers.utils.parseUnits("1000000", 6))
+    await usdc.connect(whale).transfer(borrower.address, ethers.utils.parseUnits("100000", 6))
+
+    // Lender approves USDC
+    await usdc.connect(newLender).approve(OC_ADDRESS, ethers.utils.parseUnits("100000000000", 6));
+
+    // Borrower approves USDC and borrower note
+    await usdc.connect(borrower).approve(flashRollover.address, ethers.utils.parseUnits("100000000000", 6));
+    await note.connect(borrower).approve(flashRollover.address, LOAN_ID);
+
+    console.log("Creating a vault...");
+
+    // Lender creates vault
+    const vfFactory = await ethers.getContractFactory("VaultFactory");
+    const vaultFactory = <VaultFactory>await vfFactory.attach(VAULT_FACTORY_ADDRESS);
+
+    const initTx = await vaultFactory.connect(borrower).initializeBundle(borrower.address);
+    const initReceipt = await initTx.wait();
+
+    const createdEvent = initReceipt.events?.find(e => e.event === "VaultCreated");
+    const vault = createdEvent?.args?.[0];
+
+    console.log(`Approving vault ${vault}...`);
+
+    // Lender approves vault
+    await vaultFactory.connect(borrower).approve(flashRollover.address, vault);
+
+    console.log("Creating signature...");
+
+    const loanTerms: LoanTerms = {
+        durationSecs: 7776000,
+        deadline: Math.floor(Date.now() / 1000) + 100_000,
+        numInstallments: 0,
+        interestRate: ethers.utils.parseEther("3.75"),
+        principal: repayAmount,
+        collateralAddress: VAULT_FACTORY_ADDRESS,
+        collateralId: vault,
```

Comment:
Is `collateralId` equal vault contract address ?

Reply:
Yep, ethers can take the address directly and convert it to the right type (`int256`), since the address is just the hex representation of the decimal tokenId